packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -52,12 +52,12 @@ export class RemoteChannelContext implements IChannelContext {
     /**
      * This loads the GC details from the base snapshot of this context.
      */
-    private readonly initialGCDetailsP = new LazyPromise<IGCDetails>(async () => {
+    private readonly initialGCDetailsP = new LazyPromise<IGCDetails | undefined>(async () => {
         if (await this.services.objectStorage.contains(gcBlobKey)) {
             return readAndParse<IGCDetails>(this.services.objectStorage, gcBlobKey);
         } else {
-            // Default value of initial GC details in case the initial snapshot does not have GC details blob.
-            return {};
+            // For old snapshots that do not have GC blob return undefined.
+            return undefined;
```

Comment:
Couple places where I found how result is used (like (await this.initialGCDetailsP)?.gcData) suggest that instead of allowing undefined, you could return some default value. IGCDetails.gcData is already optional, I think we can just return {used: true) from here, no? Or make used optional and return empty object.

Reply:
The only I had reason for not defaulting this to { used: true } is for the following scenario:
A document that does not have GC blob loads and when we run GC, the used state turns out to be true. If there are no other changes to this node's data, we will re-use the previous summary data since referenced used state and current used state are the same.

From correctness perspective, it would still be correct because the server will still see this as used. But from design perspective, I would think that it should have summarized this node again.
Either way works. I guess setting this to { used: true } would make the code cleaner.

Reply:
I think I also like Vlad's suggestion for initial value vs. undefined. I think if you switch to storing dependencies/routes instead of just a flag it becomes more clear as it goes from empty to something else.

Reply:
Changed it be an empty object.